services/src/api/model/db_types.rs

```diff
@@ -1539,6 +1551,411 @@ impl TryFrom<MetaDataDefinitionDbType> for MetaDataDefinition {
+            base_url: other.base_url.clone().into(),
+            overviews: other.overviews.to_string_lossy().to_string(),
+            cache_ttl: other.cache_ttl,
+        }
+    }
+}
+
+impl TryFrom<EbvPortalDataProviderDefinitionDbType> for EbvPortalDataProviderDefinition {
+    type Error = Error;
+
+    fn try_from(other: EbvPortalDataProviderDefinitionDbType) -> Result<Self, Self::Error> {
+        Ok(Self {
+            name: other.name,
+            path: other.path.into(),
+            base_url: other.base_url.as_str().try_into()?,
+            overviews: other.overviews.into(),
+            cache_ttl: other.cache_ttl,
+        })
+    }
+}
+
+#[derive(Debug, ToSql, FromSql)]
+#[postgres(name = "NetCdfCfDataProviderDefinition")]
+pub struct NetCdfCfDataProviderDefinitionDbType {
+    pub name: String,
+    pub path: String,
+    pub overviews: String,
+    pub cache_ttl: CacheTtlSeconds,
+}
+
+impl From<&NetCdfCfDataProviderDefinition> for NetCdfCfDataProviderDefinitionDbType {
+    fn from(other: &NetCdfCfDataProviderDefinition) -> Self {
+        Self {
+            name: other.name.clone(),
+            path: other.path.to_string_lossy().to_string(),
+            overviews: other.overviews.to_string_lossy().to_string(),
+            cache_ttl: other.cache_ttl,
+        }
+    }
+}
+
+impl TryFrom<NetCdfCfDataProviderDefinitionDbType> for NetCdfCfDataProviderDefinition {
+    type Error = Error;
+
+    fn try_from(other: NetCdfCfDataProviderDefinitionDbType) -> Result<Self, Self::Error> {
+        Ok(Self {
+            name: other.name,
+            path: other.path.into(),
+            overviews: other.overviews.into(),
+            cache_ttl: other.cache_ttl,
+        })
+    }
+}
+
+#[derive(Debug, ToSql, FromSql)]
+#[postgres(name = "PangaeaDataProviderDefinition")]
+pub struct PangaeaDataProviderDefinitionDbType {
+    pub name: String,
+    pub base_url: String,
+    pub cache_ttl: CacheTtlSeconds,
+}
+
+impl From<&PangaeaDataProviderDefinition> for PangaeaDataProviderDefinitionDbType {
+    fn from(other: &PangaeaDataProviderDefinition) -> Self {
+        Self {
+            name: other.name.clone(),
+            base_url: other.base_url.clone().into(),
+            cache_ttl: other.cache_ttl,
+        }
+    }
+}
+
+impl TryFrom<PangaeaDataProviderDefinitionDbType> for PangaeaDataProviderDefinition {
+    type Error = Error;
+
+    fn try_from(other: PangaeaDataProviderDefinitionDbType) -> Result<Self, Self::Error> {
+        Ok(Self {
+            name: other.name,
+            base_url: other.base_url.as_str().try_into()?,
+            cache_ttl: other.cache_ttl,
+        })
+    }
+}
+
+#[derive(Debug, PartialEq, ToSql, FromSql)]
+pub struct TextMetaDataDefinitionKeyValue {
+    key: String,
+    value: MetaDataDefinition,
+}
+
+#[derive(Debug, PartialEq, ToSql, FromSql)]
+#[postgres(transparent)]
+pub struct HashMapTextMetaDataDefinitionDbType(pub Vec<TextMetaDataDefinitionKeyValue>);
```

Comment:
this type is also redundant now, right?

Reply:
[5c7aaa7](https://github.com/geo-engine/geoengine/pull/866/commits/5c7aaa7bdbd00967ec3bff7f8e786518c9360903)